theano/tensor/elemwise.py

```diff
@@ -1221,8 +1223,9 @@ def _c_all(self, node, nodename, inames, onames, sub):
         }
         """ % locals()
 
-        loop_orders = orders + [list(range(nnested))] * len(real_onames)
-        dtypes = (idtypes + list(real_odtypes))
+        loop_orders = orders + [list(range(nnested))] * len(onames)
```

Comment:
In order to make the c code independent of the destroy maps, I'm giving all outputs even those inplace to the loop_orders variable.
To be honest I'm not sure if that's a good idea without any modifications on the functions make_loop and make_loop_reordered in elemwise_cgen.py. But the code there is a little cryptic to me and I've yet to understand it good enough to understand what problems that might generate.